src/domain/session/room/timeline/tiles/BaseMessageTile.js

```diff
@@ -45,6 +47,10 @@ export class BaseMessageTile extends SimpleTile {
         return `https://matrix.to/#/${encodeURIComponent(this._room.id)}/${encodeURIComponent(this._entry.id)}`;
     }
 
+    copyPermalink() {
+        copyPlaintext(this.permaLink);
```

Comment:
Since this code is in the domain layer, we can't import DOM code straight away here. Instead, you can expose the copy function on the platform object that is available here through `this.platform`.